vault/resource_ad_secret_backend_test.go

```diff
@@ -74,6 +74,49 @@ func TestADSecretBackend(t *testing.T) {
 	})
 }
 
+func TestADSecretBackend_remount(t *testing.T) {
+	backend := acctest.RandomWithPrefix("tf-test-ad")
+	updatedBackend := acctest.RandomWithPrefix("tf-test-ad-updated")
+
+	resourceName := "vault_ad_secret_backend.test"
+	bindDN, bindPass, url := testutil.GetTestADCreds(t)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testutil.TestAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				Config: testADSecretBackend_initialConfig(backend, bindDN, bindPass, url),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "backend", backend),
+					resource.TestCheckResourceAttr(resourceName, "description", "test description"),
+					resource.TestCheckResourceAttr(resourceName, "default_lease_ttl_seconds", "3600"),
+					resource.TestCheckResourceAttr(resourceName, "max_lease_ttl_seconds", "7200"),
+					resource.TestCheckResourceAttr(resourceName, "binddn", bindDN),
+					resource.TestCheckResourceAttr(resourceName, "bindpass", bindPass),
+					resource.TestCheckResourceAttr(resourceName, "url", url),
+					resource.TestCheckResourceAttr(resourceName, "insecure_tls", "true"),
+					resource.TestCheckResourceAttr(resourceName, "userdn", "CN=Users,DC=corp,DC=example,DC=net"),
+				),
+			},
+			{
```

Comment:
Does this really test the remount? There are two possible outcomes for the update: a new resource is created, or a graceful remount is done.

Reply:
That's a great point. So far the only definite way I found to confirm that mount migration had successfully completed was by manually checking the Vault logs and confirming that I saw the proper logging:

```
2022-09-08T10:13:34.835-0700 [INFO]  core: successful mount: namespace="" path=tf-test-terraform-cloud-5084382457978848863/ type=terraform

2022-09-08T10:13:36.012-0700 [INFO]  core.mounts.migration: Starting to update the mount table and revoke leases: from_path=tf-test-terraform-cloud-5084382457978848863 migration_id=21d4beff-97c0-afac-974c-b18daf982bfd namespace="" to_path=tf-test-terraform-cloud-updated-8145651656808162423

2022-09-08T10:13:36.012-0700 [INFO]  core.mounts.migration: Removing the source mount from filtered paths on secondaries: from_path=tf-test-terraform-cloud-5084382457978848863 migration_id=21d4beff-97c0-afac-974c-b18daf982bfd namespace="" to_path=tf-test-terraform-cloud-updated-8145651656808162423

2022-09-08T10:13:36.012-0700 [INFO]  core.mounts.migration: Updating quotas associated with the source mount: from_path=tf-test-terraform-cloud-5084382457978848863 migration_id=21d4beff-97c0-afac-974c-b18daf982bfd namespace="" to_path=tf-test-terraform-cloud-updated-8145651656808162423

2022-09-08T10:13:36.012-0700 [INFO]  core.mounts.migration: Completed mount move operations: from_path=tf-test-terraform-cloud-5084382457978848863 migration_id=21d4beff-97c0-afac-974c-b18daf982bfd namespace="" to_path=tf-test-terraform-cloud-updated-8145651656808162423

2022-09-08T10:13:38.301-0700 [INFO]  core: successfully unmounted: path=tf-test-terraform-cloud-updated-8145651656808162423/ namespace=""
```

Reply:
This test passes for versions < 1.10.0 as well, and currently the only way to know what sort of mount move happened in Vault is to check the logs. For v1.9.0 say, this test will pass in CI, but when we manually check the logs we will see that in fact a new mount was created and the previous mount was destroyed: 

```
2022-09-08T10:28:18.426-0700 [INFO]  core: successful mount: namespace="" path=tf-test-terraform-cloud-7726032830502065044/ type=terraform

2022-09-08T10:28:19.595-0700 [INFO]  core: successfully unmounted: path=tf-test-terraform-cloud-7726032830502065044/ namespace=""

2022-09-08T10:28:19.620-0700 [INFO]  core: successful mount: namespace="" path=tf-test-terraform-cloud-updated-6295110747372747410/ type=terraform

2022-09-08T10:28:20.891-0700 [INFO]  core: successfully unmounted: path=tf-test-terraform-cloud-updated-6295110747372747410/ namespace=""
```

Reply:
Unfortunately, I wasn't able to find a way for the TF SDK to know whether Vault had created a new mount and destroyed the old one, or whether Vault had remounted with a new name. Open to any suggestions :)

Reply:
Would it be possible to leverage the `uuid` of the mount, it seems that is available in the response: https://www.vaultproject.io/api-docs/system/mounts#sample-response

Reply:
Was able to set up a test for `consul_secret_backend` resource which compares UUIDs for the mounts to verify that the mount was appropriately remounted/recreated.